pydax/_dataset.py

```diff
@@ -88,6 +89,41 @@ def __init__(self, schema: SchemaDict,
         if mode & Dataset.InitializationMode.LOAD_ONLY:
             self.load()
 
+    @property
+    def _data_dir(self) -> pathlib.Path:
+        "Directory that stores datasets. Create it if not existing."
+        if not self._data_dir_.exists():
+            self._data_dir_.mkdir(parents=True)
+        elif not self._data_dir_.is_dir():  # self._data_dir_ exists and is not a directory
+            raise FileExistsError(f'"{self._data_dir_}" exists and is not a directory.')
+        return self._data_dir_
+
+    @property
+    def _cache_dir(self) -> pathlib.Path:
+        "Cache and metainfo directory used by this class. Create it if not existing."
+        cache_dir = self._data_dir / '.pydax'
```

Comment:
Sounds good that we should use a different name. This would also be easier to explain to users. I was trying to avoid making it specific to a single function. Instead we occupy this directory and our future need of various files will go here.  Does `.pydax.d` sound good to you? `d` seems to be a punchline: it is traditionally used in this way to indicate it's a  directory but here it also means dataset.

Reply:
gotcha, makes sense about not using `cache` in the name. Is there a downside to using `dataset` instead of just `d` to be more explicit. I feel like I myself will forget what the `d` stands for in like a month haha, but if you don't like `.pydax.dataset` then I'm good with `.pydax.d` too.

Reply:
Sure, we can do the full name